tests/storage/test_e2e_room_keys.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from twisted.internet import defer
+
+from tests import unittest, utils
+
+# sample room_key data for use in the tests
+room_key = {
+    "first_message_index": 1,
+    "forwarded_count": 1,
+    "is_verified": False,
+    "session_data": "SSBBTSBBIEZJU0gK",
+}
+
+
+class E2eRoomKeysHandlerTestCase(unittest.TestCase):
```

Comment:
Can you use `HomeserverTestCase` please? Its the new (ish) way of doing things such that we don't need to spin up a full reactor, and has a number of advantages.

Broadly, it works in the same way instead of doing `@defer.inlineCallbacks` and `yield` you can do `self.get_success(self.store.some_deferred_returning_func())`